gamemodes/terrortown/gamemode/shared/sh_lang.lua

```diff
@@ -24,22 +24,53 @@ util.IncludeClientFile("terrortown/gamemode/client/cl_lang.lua")
 local net = net
 local table = table
 local pairs = pairs
+local file = file
+local string = string
 
--- Add all lua files in our /lang/ dir
-local dir = GM.FolderName or "terrortown"
-local files = file.Find(dir .. "/gamemode/shared/lang/*.lua", "LUA")
+---
+-- Sets up the language by scanning through the given directory; has to be run on both
+-- server and client!
+-- @param string lang_path The path to search in
+-- @param boolean deepsearch If true, subfolders are scanned
+-- @internal
+-- @realm shared
+function LANG.SetupFiles(lang_path, deepsearch)
+	local file_paths = {}
+
+	if deepsearch then
+		local sub_folders = ({file.Find(lang_path .. "*", "LUA")})[2] or {}
```

Comment:
haha smart one :D But here isn't a need for inline things, better use: `local _, sub_folders = file.Find(lang_path .. "*", "LUA") or (nil, {})`

But why should subfolders not be an empty table after calling this function?

Reply:
I saved a `if not sub_folders then return end` with this one

Reply:
yea and doing useless stuff :D

Reply:
okay, I'll rewert it. I'll finish up the door PR first though

Reply:
ref.: https://wiki.facepunch.com/gmod/file.Find this seems to just return nil if the path is invalid, theoretically an empty table if there is no content (already). So I think this OR is useless, what do you think?

Reply:
no I tested it. `#` fails if the table is nil.

Reply:
Maybe because https://wiki.facepunch.com/gmod/file.Exists would return false. YOu don't need to call this function additionally. Your solution should work, but doing useless stuff is a bad coding way, it's unreliable. If there is something that is invalid, it should be returned. But you are right too, if the dict is valid but empty, it would so the same. So I suggest using your solution but with the improved / simplified syntax :)